### PR TITLE
feat: agents are aware of network

### PR DESCRIPTION
**Pull request by cbrzn**
updates `ResearchTokensAgent` to be aware of the network if user mentions it in the prompt
- if there's not indication of network the search will return tokens from al chains
- if user mentions network the tokens will be filtered based on that network name

it's worth mentioning that we're not doing any prompt engineering, but rather the tool `GetTokensBasedOnCategory` now accepts a `network_name` argument

also, the `SearchToken` tool has replaced `TokenSymbolToTokenId`, this makes the execution loop **way** more consistent

two tests has been added, where we ask the top 5 memecoins without specifying any chain, and another when we ask the top 5 memecoins from optimism

once #138 and this are merged, #120 can be closed

### CODE REVIEW

**Comment by cbrzn** [comment]
/workflows/benchmarks agents/token/test_token_research.py

File: autotx/tests/agents/token/test_token_research.py [hunk 1]
```diff
@@ -33,22 +33,23 @@ def test_get_token_exchanges(token_research_agent: Agent):
...
+        expected_output="Liquidity of UNI",
     )
     response = token_research_agent.execute_task(task)
-    assert "The liquidity of the UNI token" in response
+    assert "The liquidity" in response
 
 def test_get_top_5_tokens_from_base(token_research_agent: Agent):
     task = Task(
-        description="What are the top 5 tokens on Base chain?",
+        description="What are the top 5 tokens on Base network?",
```
Comment (nerfZael) [inline autotx/tests/agents/token/test_token_research.py:50]: Hmm, I think it's a valid use case for the user to use the `chain` wording

File: autotx/agents/ResearchTokensAgent.py [hunk 2]
```diff
@@ -87,67 +93,114 @@ def _run(self, token_id: str):
             }
         )
 
+
+class SearchToken(BaseTool):
+    name: str = "search_token"
+    description: str = dedent(
+        """
+        Search token based on its symbol. It will return the ID of tokens with most market cap
```
Comment (nerfZael) [inline autotx/agents/ResearchTokensAgent.py:101]: `with the largest market cap`?

File: autotx/agents/ResearchTokensAgent.py [hunk 3]
```diff
@@ -87,67 +93,114 @@ def _run(self, token_id: str):
...
+
+        Args:
+            token_symbol (str): Symbol of token to search
+            retrieve_all (bool): Only used to retrieve all tokens (Which can happen if they share the same symbol).
```
Comment (nerfZael) [inline autotx/agents/ResearchTokensAgent.py:105]: Maybe name this `retrieve_duplicates`?

**Comment by cbrzn** [comment]
/workflows/benchmarks agents/token/test_token_research.py

File: autotx/agents/ResearchTokensAgent.py [hunk 4]
```diff
@@ -87,67 +93,114 @@ def _run(self, token_id: str):
...
+                By default use False unless it's necessary to use True
+        """
+    )
+
+    def _run(self, token_symbol: str, retrieve_all: str):
```
Comment (nerfZael) [inline autotx/agents/ResearchTokensAgent.py:110]: Shouldn't `retrieve_all` be bool?

File: autotx/agents/ResearchTokensAgent.py [hunk 5]
```diff
@@ -87,67 +93,114 @@ def _run(self, token_id: str):
...
 class GetTokensBasedOnCategory(BaseTool):
     name: str = "get_tokens_based_on_category"
     description: str = dedent(
-        """
-        Retrieve all tokens from a given category
+        f"""
+        Retrieve all tokens with their respective information (id, symbol, market cap, price change percentages and total traded volume in the last 24 hours) from a given category
 
         Args:
             category (str): Category to retrieve tokens
```
Comment (nerfZael) [inline autotx/agents/ResearchTokensAgent.py:140]: What is a category?

File: autotx/agents/ResearchTokensAgent.py (hunk 4 above)
Comment (cbrzn) [reply autotx/agents/ResearchTokensAgent.py:110]: yes it should but it seems that the LLM always pass the argument as string even tho i explicitly mentioned it to be a boolean :disappointed:

File: autotx/agents/ResearchTokensAgent.py (hunk 3 above)
Comment (cbrzn) [reply autotx/agents/ResearchTokensAgent.py:105]: yah i like this more

**Review by nerfZael — APPROVED** [review]

**Pull request merged**